lib/grppcatr.gi

```diff
@@ -466,6 +466,49 @@ end );
 
 #############################################################################
 ##
+#M  MaximalNormalSubgroups( <G> )
+##
+InstallMethod( MaximalNormalSubgroups, "for solvable groups",
+              [ IsGroup ],
+              RankFilter( IsGroup and IsSolvableGroup )
+              - RankFilter( IsGroup ),
+
+  function( G )
+    local Gf,     # FactorGroup of G
+          hom,    # homomorphism from G to Gf
+          MaxGf;  # MaximalNormalSubgroups of Gf
+
+    if 0 in AbelianInvariants(G) then
+      # (p) is a maximal normal subgroup in Z for every prime p
+      Error("number of maximal normal subgroups is infinity");
+    elif IsAbelian(G) then
+      if not IsPcGroup(G) then
+        # convert it to an Abelian PcGroup with same invariants
+        Gf := AbelianGroup(IsPcGroup, AbelianInvariants(G));
+        hom := IsomorphismGroups(G, Gf);
+        MaxGf := MaximalNormalSubgroups(Gf);
+        return List(MaxGf, N -> PreImage(hom, N));
+      else
+        # for abelian pc groups return all maximal subgroups
+        # NormalMaximalSubgroups seems to omit some unnecessary checks,
+        # hence faster than MaximalSubgroups
+        return NormalMaximalSubgroups(G);
+      fi;
```

Comment:
If `NormalMaximalSubgroups` is faster, then why don't you use it above, on Gf, which is a pc group, too?